drivers/Aqara/aqara-lock/src/init.lua

```diff
@@ -78,11 +108,19 @@ end
 
 local function event_unlock_handler(driver, device, evt_name, evt_value)
   local id, label
-  if evt_value == 0x80020000 then -- one-time password
+  if evt_name == METHOD.RF447 then
+    evt_name = nil
+    id = nil
+    label = nil
+  elseif evt_name == METHOD.BLUETOOTH and evt_value == 2 then
+    evt_name = nil
+    id = nil
+    label = nil
+  elseif evt_value == 0x80020000 then -- one-time password
     id = "OTP_STANDALONE"
     label = nil
   else
-  id, label = credential_utils.find_userLabel(driver, device, evt_value)
+    id, label = credential_utils.find_userLabel(driver, device, evt_value)
   end
   device:emit_event(Lock.lock.unlocked({ data = { method = evt_name, codeId = id, codeName = label } }))
```

Comment:
Do you still want to emit the event even when `evt_name` is nil? Or maybe still emit the event but without `data`?

Reply:
There are no cases where `evt_name` is nil in the code, but should I add code to double-check it?

Reply:
Lines 113 and 117 are setting `evt_name` to `nil` so yes I think there should be code to protect against it

Reply:
I set that part to “nil” at Samsung's request.

Reply:
@tpmanley, Actually I asked Aqara team not to send `codeId`, `codeName`, `method` in case of BLUETOOTH, RF447 which was intended "emit the event but without `data`" just like you mentioned.
Do you think we need to seprate `emit_event` implementation?

Reply:
@tpmanley , we have updated the `lock` capability to have more `method` types. So I think it is better to fill the method as `proprietaryRemote` with `nil` of `codeId`, `codeName` in case of bluetooth and RF447 use cases.
Could you let me know your opinion?